app/controllers/user_profiles_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Controller for create and edit UserProfile
+class UserProfilesController < ApplicationController
+  before_action :set_major, only: %i[new edit]
+  # before_action ...
+
+  def new
+    @user = User.new
+    # @user_profile = current_user.build_user_profile
+    @user_profile = @user.build_user_profile
+  end
+
+  def create
+    @user = User.new
+    @user.save
+    # @user_profile = current_user.build_user_profile(profile_params)
+    @user_profile = @user.build_user_profile(profile_params)
+    @user_profile.save
+    # if @user_profile.save
+    #   render '...'
+    # else
+    #   render 'new'
+    # end
```

Comment:
詳細ページがないので保存に成功した場合の遷移先は書けないと思いますが、バリデーションに失敗した場合に `new` に戻す処理はコメントアウトされてないほうが良いかもと思いました。

```suggestion
    if @user_profile.save
    #   render '...'
    else
      render 'new'
    end
```

Reply:
保存が成功するとき、暫くeditへ飛ばします。失敗の時、newに戻すようになりました。ありがとうがざいます。